pkg/api/v1alpha1/sys_extension_resources.go

```diff
@@ -0,0 +1,702 @@
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := updateContextWithAuditEventData(c, event); err != nil {
+		msg := fmt.Sprintf("error creating extension resource: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	if err := tx.Commit(); err != nil {
+		msg := fmt.Sprintf("error committing extension resource create: %s", err.Error())
+
+		if err := tx.Rollback(); err != nil {
+			msg += fmt.Sprintf("error rolling back transaction: %s", err.Error())
+		}
+
+		sendError(c, http.StatusBadRequest, msg)
+
+		return
+	}
+
+	err = r.EventBus.Publish(
+		c.Request.Context(),
+		erd.SlugPlural,
+		&events.Event{
+			Version:                       erd.Version,
+			Action:                        events.GovernorEventCreate,
+			AuditID:                       c.GetString(ginaudit.AuditIDContextKey),
+			ActorID:                       getCtxActorID(c),
+			ExtensionID:                   extension.ID,
+			ExtensionResourceID:           er.ID,
+			ExtensionResourceDefinitionID: erd.ID,
+		},
+	)
+	if err != nil {
+		sendError(
+			c,
+			http.StatusBadRequest,
+			fmt.Sprintf(
+				"failed to publish extension resource create event: %s\n%s",
+				err.Error(),
+				"downstream changes may be delayed",
+			),
+		)
+
+		return
+	}
+
+	resp := &SystemExtensionResource{
+		SystemExtensionResource: er,
+		ERD:                     erd.SlugSingular,
+		Version:                 erd.Version,
+	}
+
+	c.JSON(http.StatusCreated, resp)
+}
+
+// listSystemExtensionResource lists system extension resources for an ERD
+func (r *Router) listSystemExtensionResources(c *gin.Context) {
+	extensionSlug := c.Param("ex-slug")
+	erdSlugPlural := c.Param("erd-slug-plural")
+	erdVersion := c.Param("erd-version")
+
+	// find ERD
+	_, erd, err := findERDForExtensionResource(
+		c.Request.Context(), r.DB,
+		extensionSlug, erdSlugPlural, erdVersion,
+	)
+	if err != nil {
+		if errors.Is(err, ErrExtensionNotFound) || errors.Is(err, ErrERDNotFound) {
+			sendError(c, http.StatusNotFound, err.Error())
+			return
+		}
+
+		sendError(c, http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	if erd.Scope != ExtensionResourceDefinitionScopeSys.String() {
+		sendError(
+			c, http.StatusBadRequest,
+			fmt.Sprintf(
+				"cannot list system resources for %s scoped %s/%s",
+				erd.Scope, erd.SlugSingular, erd.Version,
+			),
+		)
+
+		return
+	}
+
+	uriQueries := map[string]string{}
+	if err := c.BindQuery(&uriQueries); err != nil {
+		sendError(
+			c, http.StatusBadRequest,
+			fmt.Sprintf("error binding uri queries: %s", err.Error()),
+		)
+
+		return
+	}
+
+	qms := make([]qm.QueryMod, 0, len(uriQueries))
+
+	for k, v := range uriQueries {
+		if k == "deleted" {
+			qms = append(qms, qm.WithDeleted())
+			continue
+		}
+
+		qms = append(qms, qm.Where("resource->? = ?", k, v))
```

Comment:
I think this might have to be `->>` to work as intended. Not sure how/if binding the variables in like this instead of just writing the query out by hand affects things, but see the db-fiddle example I came up with below

```suggestion
		qms = append(qms, qm.Where("resource->>? = ?", k, v))
```

[https://www.db-fiddle.com/f/4jyoMCicNSZpjMt4jFYoz5/10523](https://www.db-fiddle.com/f/4jyoMCicNSZpjMt4jFYoz5/10523)